// canonical/dist/bin/commands/fixCommand.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _yargs = require('yargs');

var _yargs2 = _interopRequireDefault(_yargs);

var _getStdin = require('get-stdin');

var _getStdin2 = _interopRequireDefault(_getStdin);

var _utilities = require('./../../utilities');

var _bin = require('./../../utilities/bin');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var handleStdin = function handleStdin(argv) {
    /* eslint-disable no-unused-expressions */
    _yargs2.default
    /* eslint-enable */

    /* eslint-enable */
    .demand(0, 0).argv;

    (0, _getStdin2.default)().then(function (stdin) {
        var result = (0, _utilities.fixText)(stdin, {
            syntax: argv.syntax
        });

        /* eslint-disable no-console */
        console.log(result);
        /* eslint-enable */
    });
};

var handleFilePaths = function handleFilePaths(argv) {
    /* eslint-disable no-unused-expressions */
    _yargs2.default.demand(1).argv;

    var targetPaths = (0, _bin.getTargetPaths)(argv._);

    var report = (0, _utilities.fixFiles)(targetPaths);

    /* eslint-disable no-console */
    console.log(report);
    /* eslint-enable */
};

exports.default = function (subYargs) {
    subYargs.exitProcess(false).fail(_bin.failHandler);

    var argv = (0, _yargs2.default)(process.argv.splice(3)).exitProcess(false).fail(_bin.failHandler).help('help').strict().options({
        stdin: {
            default: false,
            describe: 'Used to indicate that subject body will be read from stdin.',
            type: 'boolean'
        },
        syntax: {
            // @todo This need to be true when using stdin.
            choices: ['css', 'js', 'json', 'scss'],
            demand: false,
            describe: 'Syntax of the input.'
        }
    }).argv;

    if (argv.stdin) {
        handleStdin(argv);
    } else {
        handleFilePaths(argv);
    }
};

module.exports = exports['default'];
